keymanagementscenarios.md

```diff
@@ -0,0 +1,187 @@
+- Trust Store Key Information: The key information configured in the trust store will be cryptographically verifiable and contain:
+    - Public Key: Will be used to verify signed artifacts.
+    - Signed Identity: Identity of creator of the signing key.
+    - (Optional) Issuer: If the key is an intermediate this will describe the owner of the root.
+    - Mechanism to update validity of signatures generated with the signing key.
+
+## Requirements
+- Signing an artifact SHOULD NOT require the publisher to perform additional actions with a registry/repository or registry operator beyond those required to push an unsigned artifact.
+- Validating a signature SHOULD NOT require the deployer to perform additional actions with a registry/repository or registry operator beyond those required to pull an unsigned artifact.
+- Moving an artifact from one repository to another SHOULD NOT invalidate the signature on the artifact.
+- Publishers SHOULD be able to sign with keys stored on their local machines, secure tokens, Hardware Security Modules (HSMs), or cloud based Key Management Services.
+- Publishers SHOULD be able to generate multiple signatures for a single artifact.
+- Publishers MUST have a mechanism to revoke signatures to indicate they are no longer trusted. 
+- Trust stores SHOULD be configurable by the deployer.
+- Deployers MUST be able to configure trusted entities for individual repositories and targets.
+- Deployers MUST be able to validate signatures on any version of an artifact including whether they have been revoked by the publisher.
+- Signature validation MUST be enforceable in air-gapped environments. 
+
+## Requirements for discussion
+- Providing a mechanism for rotating a root key: In prior discussions we did not close on whether Notary v2 should provide a mechanism for rotating a root key or not. Tradeoffs are listed below for further discussion.
+    - Adding a mechanism:
+        - Allows for the automatic rotation of keys. Can be used to mass update keys if signing algorithms need to be updated.
+        - Doesn't address the use case of publishers losing a root key.
+        - Provides a bad actor mechanism to insert a new root without deployer acknowledgement.
+    - Not adding a mechanism:
+        - Any rotation of root keys will require deployers to acknowledge and make a change.
+        - Doesn't address the use case of publishers losing a root key.
+        - Prevents a bad actor from inserting a new root without deployer acknowledgement.
```

Comment:
While this prevents new root keys from being created, an attacker could create any number of other delegated keys and sign any artifacts they want with a compromised root key. At that point the added security from not allowing root key rotation seems minimal.